src/mlpack/core/util/param_checks_impl.hpp

```diff
@@ -306,6 +306,51 @@ inline void ReportIgnoredParam(util::Params& params,
   }
 }
 
+// Check if the given input data points aren't empty
+template<typename T>
+inline void RequireNonEmptyInputValue(
+    util::Params& params,
+    const std::string& paramName,
+    const bool fatal,
+    const std::string& errorMessage)
+{
+  if (BINDING_IGNORE_CHECK(paramName))
+    return;
+
+  if (params.Has(paramName))
+  {
+    mat trainingData = std::move(params.Get<arma::mat>(paramName));
+    if (trainingData.empty())
+    {
+      util::PrefixedOutStream& stream = fatal ? Log::Fatal : Log::Warn;
+      if (!errorMessage.empty())
+        stream << paramName << errorMessage << std::endl;
+    }
+  }
+}
+
+inline void RequireNonEmptyInputValue(
+    util::Params& params,
+    const std::string& paramName,
+    const std::pair<std::string, bool>& constraints,
+    const bool fatal,
+    const std::string& errorMessage)
+{
+  if (BINDING_IGNORE_CHECK(paramName))
+    return;
+
+  if (params.Has(paramName) && constraints.second)
+  {
+    mat trainingData = std::move(std::get<1>(params.Get<TupleType>(paramName)));
+    if (trainingData.empty())
+    {
+      util::PrefixedOutStream& stream = fatal ? Log::Fatal : Log::Warn;
+      if (!errorMessage.empty())
+        stream << paramName << errorMessage << std::endl;
+    }
+  }
+}
```

Comment:
Can you declare its type in `param_checks.hpp`?
And what is `TupleType`?
And Why do you need constraints object?

Reply:
I didn't it. What do you mean by defining its type? constraints parameter is for checking if the regressor or classifier is a tree and perform checks likewise.